stdlib/REPL/src/REPL.jl

```diff
@@ -66,20 +66,26 @@ mutable struct REPLBackend
     response_channel::Channel
     "flag indicating the state of this backend"
     in_eval::Bool
+    "transformation functions to apply before evaluating expressions"
+    ast_transforms::Vector{Any}
     "current backend task"
     backend_task::Task
 
-    REPLBackend(repl_channel, response_channel, in_eval) =
-        new(repl_channel, response_channel, in_eval)
+    REPLBackend(repl_channel, response_channel, in_eval, ast_transforms=copy(repl_ast_transforms)) =
+        new(repl_channel, response_channel, in_eval, ast_transforms)
 end
 
-function softscope!(ex)
+"""
+    softscope(ex)
+
+Return a modified version of the parsed expression `ex` that uses
+the REPL's "soft" scoping rules for global syntax blocks.
+"""
+function softscope(@nospecialize ex)
     if ex isa Expr
         h = ex.head
         if h === :toplevel
-            for i = 1:length(ex.args)
-                ex.args[i] = softscope!(ex.args[i])
-            end
+            return Expr(h, Any[softscope(a) for a in ex.args])
```

Comment:
```
e = Expr(h)
e.args = Any[softscope(a) for a in ex.args]
return e
```
Somewhat strangely, I don't believe there is a standard non-varargs Expr constructor.